src/main/java/bookstore/controller/OrderController.java

```diff
@@ -0,0 +1,54 @@
+package bookstore.controller;
+
+import bookstore.dto.order.OrderDto;
+import bookstore.dto.order.OrderItemDto;
+import bookstore.service.order.OrderService;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/orders")
+@RequiredArgsConstructor
+public class OrderController {
+    private OrderService orderService;
+
+    @PreAuthorize("hasRole('ADMIN')")
+    @PutMapping("/{id}")
+    public OrderDto updateStatus(@PathVariable Long id, @RequestBody String status) {
```

Comment:
Controller may easily deserialize into Enum
```suggestion
    public OrderDto updateStatus(@PathVariable Long id, @RequestBody Status status) {
```